### PR TITLE
remove height field from the endorsement

### PR DESCRIPTION
**Pull request by Mr-Leshiy**

### CODE REVIEW

File: include/veriblock/blockchain/commands/addendorsement.hpp [hunk 1]
```diff
@@ -48,19 +48,13 @@ struct AddEndorsement : public Command {
...
-      return state.Invalid(protected_block_t::name() + "-expired",
-                           "Endorsement expired");
-    }
-
-    auto* endorsed = chain[endorsedHeight];
+    auto* endorsed = ed_->getBlockIndex(e_->endorsedHash);
     if (!endorsed) {
       return state.Invalid(protected_block_t::name() + "-no-endorsed-block",
-                           "No block found on endorsed block height");
+                           "No block found the tree");
```
Comment (blackyblack) [inline include/veriblock/blockchain/commands/addendorsement.hpp:54]: "Endorsed block not found"

File: include/veriblock/blockchain/commands/addendorsement.hpp [hunk 2]
```diff
@@ -48,19 +48,13 @@ struct AddEndorsement : public Command {
...
     }
 
-    if (endorsed->getHash() != e_->endorsedHash) {
+    if (endorsed->getHash() != chain[endorsed->height]->getHash()) {
```
Comment (blackyblack) [inline include/veriblock/blockchain/commands/addendorsement.hpp:57]: Need additional check that chain[endorsed->height] exists. It seems you've just removed this check.

Comment (Mr-Leshiy) [reply include/veriblock/blockchain/commands/addendorsement.hpp]: I have fixed it

File: include/veriblock/blockchain/commands/addendorsement.hpp (hunk 1 above)
Comment (Mr-Leshiy) [reply include/veriblock/blockchain/commands/addendorsement.hpp]: fixed

**Pull request merged**